### PR TITLE
41x4dyt

### PR DESCRIPTION
**Pull request by YuppiePF**
## Issue ticket number and link

https://app.clickup.com/t/41x4dyt

## Description

Decouple exporter services from monitoring services.

## Type of change

Please delete option that is not relevant.

- [X] New feature (non-breaking change which adds functionality)

### CODE REVIEW

**Comment by tohoPF** [comment]
We shouldn't assume root. Lets use default docker group and update file permissions. It would be a good idea to also include chown and chmod steps in the setup.py ..

File: clients/bcexporter/config/Config.py [hunk 1]
```diff
@@ -15,5 +14,5 @@ def _get_config(self):
             return yaml.safe_load(stream)
 
     def _get_chains(self):
-        with open('./config/chains.json', "r") as stream:
+        with open("./config/chains.json", "r") as stream:
```
Comment (tohoPF) [inline clients/bcexporter/config/Config.py:17]: Let's use[ pathlib ](https://docs.python.org/3/library/pathlib.html)

File: templates/clients/docker-compose.yml [hunk 2]
```diff
@@ -8,6 +8,9 @@ services:
   blockchain_exporter:
     container_name: blockchain_exporter
     image: nodies/blockchain_exporter
+    user: "0:0"
```
Comment (tohoPF) [inline templates/clients/docker-compose.yml]: remove hardcoded root user and change file permissions to use the docker group. add a step in setup.py that does this.

File: clients/bcexporter/config/Config.py (hunk 1 above)
Comment (tohoPF) [inline clients/bcexporter/config/Config.py:17]: Let's use pathlib

File: variables.py [hunk 3]
```diff
@@ -67,7 +67,7 @@
     for source in ['loki', 'prometheus']:
         with open(f'{template_datasource}/{source}.yaml', "r") as f:
             newdict = yaml.safe_load(f)
-        newdict["datasources"][0]["url"] = 'https://' + os.getenv("MONITORING_ENDPOINT") + ':' + os.getenv(f'{source.upper()}_PORT')
+        newdict["datasources"][0]["url"] = 'http://' + os.getenv("MONITORING_ENDPOINT") + ':' + os.getenv(f'{source.upper()}_PORT')
```
Comment (tohoPF) [inline variables.py]: use urllib

File: variables.py [hunk 4]
```diff
@@ -81,7 +81,7 @@
 if run_client_promtail:
     with open(template_client_promtail, "r") as f:
         newdict = yaml.safe_load(f)
-    newdict["clients"][0]["url"] = 'https://' + os.getenv('MONITORING_ENDPOINT') + ':' + os.getenv('LOKI_PORT') + '/loki/api/v1/push'
+    newdict["clients"][0]["url"] = 'http://' + os.getenv('MONITORING_ENDPOINT') + ':' + os.getenv('LOKI_PORT') + '/loki/api/v1/push'
```
Comment (tohoPF) [inline variables.py]: use urllib

File: variables.py [hunk 5]
```diff
@@ -81,7 +81,7 @@
 if run_client_promtail:
     with open(template_client_promtail, "r") as f:
         newdict = yaml.safe_load(f)
-    newdict["clients"][0]["url"] = os.getenv('MONITORING_ENDPOINT') + ':' + os.getenv('LOKI_PORT') + '/loki/api/v1/push'
+    newdict["clients"][0]["url"] = 'https://' + os.getenv('MONITORING_ENDPOINT') + ':' + os.getenv('LOKI_PORT') + '/loki/api/v1/push'
```
Comment (tohoPF) [inline variables.py]: use urllib

**Review by tohoPF — CHANGES_REQUESTED** [review]

File: clients/bcexporter/config/Config.py (hunk 1 above)
Comment (nodiesBlade) [inline clients/bcexporter/config/Config.py:17]: Is chains.json mounted yet in docker-compose? It should not be expected that chains.json is part of the python build, given it's a config file that should be editable and changeable

File: variables.py [hunk 6]
```diff
@@ -1,13 +1,17 @@
 import yaml
```
Comment (nodiesBlade) [inline variables.py]: Whole file needs to be cleaned up. Variables.py should be renamed to setup.py

Furthermore, there should be clear, concise code of each setup component, and there is also DRY being broken here with how we load and replace files in general.

Script like coding is fine, but if we're going to do it in a high level language, let's take advantage of it.

File: variables.py [hunk 7]
```diff
@@ -74,15 +81,15 @@
...
         newdict = yaml.safe_load(f)
-    newdict["clients"][0]["url"] = os.getenv('LOKI_ENDPOINT') + ':' + os.getenv('LOKI_PORT') + '/loki/api/v1/push'
-    newdict["server"]["http_listen_port"] = os.getenv('PROMTAIL_PORT')
+    newdict["clients"][0]["url"] = 'http://' + os.getenv('MONITORING_ENDPOINT') + ':' + os.getenv('LOKI_PORT') + '/loki/api/v1/push'
+    newdict["server"]["http_listen_port"] = int(os.getenv('PROMTAIL_PORT'))
     with open(output_client_promtail, "w") as f:
         yaml.dump(newdict, f)
 
 if run_client_bcexporter:
     with open(template_client_bcexporter, "r") as f:
```
Comment (nodiesBlade) [inline variables.py]: example of DRY when loading files. just to replace a yaml var

File: templates/clients/docker-compose.yml [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+      - "^/(sys|proc|dev|host|etc|rootfs/var/lib/docker/containers|rootfs/var/lib/docker/overlay2|rootfs/run/docker/netns|rootfs/var/lib/docker/aufs)($$|/)"
+    ports:
+      - "9100:9100"
+    restart: always
+    networks:
+      clients:
+    deploy:
+      mode: global
+
+#  cadvisor:
```
Comment (nodiesBlade) [inline templates/clients/docker-compose.yml]: Why did we remove cadvisor?

**Pull request closed without merging**